src/components/PreviewFrame.jsx

```diff
@@ -100,6 +110,13 @@ class PreviewFrame extends React.Component {
     });
   }
 
+  _postHighlighterSelectorToFrame(selector) {
+    window.frames[0].postMessage(JSON.stringify({
```

Comment:
Generally a React component should never make any assumptions about the structure of the DOM outside of that which it directly controls (and try to limit assumptions even of the DOM it does control). So, the preferred way to handle this is to add a [`ref` handler](https://facebook.github.io/react/docs/refs-and-the-dom.html) to the `<iframe>`, and then store it as a property of the component instance (e.g. `this._frame_element`).

Reply:
This makes sense. I added the ref but had to add allow-same-origin. Is this okay?

Reply:
I’m a frayed knot—this opens up a security hole where an attacker could send a snapshot or gist that attempts to break out of the preview iframe and hijack Popcode’s access to the coder’s GitHub account. As far as I know, though, setting up a `ref` should behave exactly the same as `window.frames[0]` here—what happened that spurred you to add `allow-same-origin`?

Reply:
@pwjablonski just checked out the branch and removed the `allow-same-origin`—seems to work fine?

Reply:
I figured it out.
I had `console.log(this._frame_element.contentWindow);`
Which gave the error "Blocked a frame with origin "http://localhost:3000" from accessing a cross-origin frame"
Just tried it again without allow-same-origin and it works just fine. Sorry for the confusion.